#!/usr/bin/env python
PACKAGE = "my_local_planner"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("position_tolerance", double_t, 0, "Goal position tolerance", 0.05, 0.0, 1) #default 5cm
gen.add("orientation_tolerance",	double_t, 0, "Goal orientation tolerance", 0.1, 0.0, 3.14) #default 6 degrees

gen.add("lidar_detection_range_left", int_t, 0, "number of scan samples to check for obstacles", 60, 0, 180) #default 60 degrees
gen.add("lidar_detection_range_right",	int_t, 0, "number of scan samples to check for obstacles", 300, 181, 360) #default -60 degrees
gen.add("stopping_distance", double_t, 0, "distance for emergency stop", 0.5, 0.0, 1.0) # [m]

gen.add("local_plan_horizon", int_t, 0, "Temporary target global waypoint for Local Planner", 50, 1, 200)

gen.add("max_vel_lin", double_t, 0, "Max linear velocity", 0.2, 0.0, 0.6)
gen.add("min_vel_lin", double_t, 0, "Min linear velocity", 0.0, 0.0, 1.0)
gen.add("max_incr_lin", double_t, 0, "Max linear acceleration", 0.1, 0.0, 0.5)

gen.add("max_vel_ang", double_t, 0, "Max angular velocity", 0.4, 0.0, 0.5)
gen.add("min_vel_ang", double_t, 0, "Min angular velocity", -0.4, -1.0, 0.0)
gen.add("max_incr_ang", double_t, 0, "Max angular acceleration", 0.25, 0.0, 5)

gen.add("k_p_lin", double_t, 0, "Proportional gain for linear velocity", 2, 0.0, 10.0)
gen.add("k_i_lin", double_t, 0, "Integral gain for linear velocity", 0.04, 0.0, 2.0)
gen.add("k_d_lin", double_t, 0, "Derivative gain for linear velocity", 0, 0.0, 10.0)

gen.add("k_p_ang", double_t, 0, "Proportional gain for angular velocity", 2, 0.0, 10.0)
gen.add("k_i_ang", double_t, 0, "Integral gain for linear velocity", 0.0, 0.0, 2.0)
gen.add("k_d_ang", double_t, 0, "Derivative gain for linear velocity", 0.0, 0.0, 10.0)

exit(gen.generate(PACKAGE, "my_local_planner", "MyLocalPlanner"))